x-pack/plugins/rollup/server/routes/api/indices.js

```diff
@@ -37,4 +37,63 @@ export function registerIndicesRoute(server) {
       }
     }
   });
+
+  /**
+   * Returns information on validiity of an index pattern for creating a rollup job:
+   *  - Does the index pattern match any indices?
+   *  - Does the index pattern match rollup indices?
+   *  - Which time fields are available in the matching indices?
+   */
+  server.route({
+    path: '/api/rollup/index_pattern_validity/{indexPattern}',
+    method: 'GET',
+    config: {
+      pre: [ licensePreRouting ]
+    },
+    handler: async (request, reply) => {
+      const callWithRequest = callWithRequestFactory(server, request);
+
+      try {
+        const { indexPattern } = request.params;
+        const [ fieldCapabilities, rollupIndexCapabilities ] = await Promise.all([
+          callWithRequest('rollup.fieldCapabilities', { indexPattern }),
+          callWithRequest('rollup.rollupIndexCapabilities', { indexPattern }),
+        ]);
+
+        // If there are no field capabilities then the API will return a 404, and we will end up
+        // in the error handler. So if we're at this point, then we know we have matching indices.
+        const doesMatchIndices = true;
+        const doesMatchRollupIndices = Object.entries(rollupIndexCapabilities).length !== 0;
+
+        const fieldCapabilitiesEntries = Object.entries(fieldCapabilities.fields);
+        const timeFields = fieldCapabilitiesEntries.reduce((accumulatedTimeFields, [ fieldName, fieldCapability ]) => {
+          if (fieldCapability.date) {
+            accumulatedTimeFields.push(fieldName);
+          }
+          return accumulatedTimeFields;
+        }, []);
+
+        reply({
```

Comment:
I do not have any indices containing `test` in the name. when I type just `test`, the response is correct (matches no indices):
![image](https://user-images.githubusercontent.com/1965714/45240509-cb95c580-b29d-11e8-87fb-14bc7a1f3bb7.png)

but when I type `test*`, response says matching indices but no time fields:
![image](https://user-images.githubusercontent.com/1965714/45240494-bf116d00-b29d-11e8-945c-58b5dc305601.png)

Reply:
Good catch!